packages/scripts/scripts/format-js.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+const { sync: spawn } = require( 'cross-spawn' );
+const { sync: resolveBin } = require( 'resolve-bin' );
+const { sync: dirGlob } = require( 'dir-glob' );
+
+/**
+ * Internal dependencies
+ */
+const {
+	fromConfigRoot,
+	fromProjectRoot,
+	getArgFromCLI,
+	getFileArgsFromCLI,
+	hasArgInCLI,
+	hasPackageProp,
+	hasProjectFile,
+} = require( '../utils' );
+
+// See: https://prettier.io/docs/en/configuration.html
+const hasProjectPrettierConfig =
+	hasProjectFile( '.prettierrc' ) ||
+	hasProjectFile( '.prettierrc.json' ) ||
+	hasProjectFile( '.prettierrc.yaml' ) ||
+	hasProjectFile( '.prettierrc.yml' ) ||
+	hasProjectFile( '.prettierrc.js' ) ||
+	hasProjectFile( '.prettierrc.config.js' ) ||
+	hasProjectFile( '.prettierrc.toml' ) ||
+	hasPackageProp( 'prettier' );
+
+// Get the configuration file, with preference highest to lowers:
+// 1. use the `--config` arg from CLI
+// 2. use configuration provided by the project
+// 3. use the default provided in the scripts module
+let configArgs;
+const configFromCLI = getArgFromCLI( '--config' );
```

Comment:
It looks like params passed to `prettier` are whitelisted. Interesting approach. In other places, we would pass whatever the users provide:
https://github.com/WordPress/gutenberg/blob/9783474bdd86df5b82137663bc2e7fec56c7d779/packages/scripts/scripts/lint-js.js#L19-L50

I'm just sharing for the full picture. It works for me the way it's implemented as well. It's more effort probably the way I see it here :)

Reply:
I know I know, I spent some considerable time tuning the CLI args 🙂 The reason is that I can't forward all CLI arguments verbatim, because I need to change some of them.

Specifically, I need to convert the "file args" from directory names to globs (`packages/compose` to `packages/compose/**/*.js`) with the `dir-glob` package. And the CLI utils module doesn't provide helpers to separate the file args from regular args and forward only the regular ones.

The `lint-js` script always forwards all args without modification, it only adds some defaults when they are not present.